cirq-core/cirq/devices/superconducting_qubits_noise_properties.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass, field
+from typing import Dict, TYPE_CHECKING, List, Set
+
+from cirq import ops, protocols, devices
+from cirq.devices.noise_utils import (
+    OpIdentifier,
+    decoherence_pauli_error,
+)
+
+if TYPE_CHECKING:
+    import cirq
+
+SINGLE_QUBIT_GATES: Set[type] = {
+    ops.ZPowGate,
+    ops.PhasedXZGate,
+    ops.MeasurementGate,
+    ops.ResetChannel,
+}
+SYMMETRIC_TWO_QUBIT_GATES: Set[type] = {
+    ops.FSimGate,
+    ops.PhasedFSimGate,
+    ops.ISwapPowGate,
+    ops.CZPowGate,
+}
+ASYMMETRIC_TWO_QUBIT_GATES: Set[type] = set()
+TWO_QUBIT_GATES = SYMMETRIC_TWO_QUBIT_GATES | ASYMMETRIC_TWO_QUBIT_GATES
+
+
+# TODO: missing per-device defaults
+@dataclass
+class SuperconductingQubitsNoiseProperties(devices.NoiseProperties):
+    """Noise-defining properties for a quantum device.
+
+    Args:
+        gate_times_ns: Dict[type, float] of gate types to their duration on
+            quantum hardware.
+        t1_ns: Dict[cirq.Qid, float] of qubits to their T_1 time, in ns.
+        tphi_ns: Dict[cirq.Qid, float] of qubits to their T_phi time, in ns.
```

Comment:
If you have T1 then indeed T2 and Tphi can be determined from each other. I used Tphi because in the actual implementation the two terms contributing to the master equation are proportional to (1/ T1) and (1/Tphi). Presumably T2 is quoted more often because it is measured directly, whereas Tphi is usually  inferred from T1 and T2 data. So in terms of the user-facing code, I suppose it makes more sense to use T2 and then convert to Tphi in the underlying implementation.

Reply:
On second thought, here is another point. Suppose I want to include only T1 decay and no white noise dephasing. In this case I would still need to specify a dictionary of T2 even though I know it must be equal to (2 T1). With the above implementation, I could just specify the T1 dictionary and have the Tphi dictionary remain empty.